pkg/controller/cloud/node_controller.go

```diff
@@ -243,6 +250,24 @@ func (cnc *CloudNodeController) MonitorNode() {
 		// from the cloud provider. If node cannot be found in cloudprovider, then delete the node immediately
 		if currentReadyCondition != nil {
 			if currentReadyCondition.Status != v1.ConditionTrue {
+				// we need to check this first to get taint working in similar in all cloudproviders
+				// current problem is that shutdown nodes are not working in similar way ie. all cloudproviders
+				// does not delete node from kubernetes cluster when instance it is shutdown see issue #46442
+				shutdown, err := instances.InstanceShutdownByProviderID(context.TODO(), node.Spec.ProviderID)
```

Comment:
Could we reuse the logic in this util here? `pkg/controller/util/node/controller_utils.go`. I just noticed a bug in `ExistsInCloudProvider` because we didn't reuse the same code across controllers I need to open a PR for.